ui/apps/platform/src/Containers/SystemConfig/Details/PrivateConfigDataRetentionDetails.tsx

```diff
@@ -13,18 +13,20 @@ type DataRetentionValueProps = {
     value: number | undefined;
     suffix: string;
     shouldPluralize?: boolean;
+    canRetainForever?: boolean;
 };
 
 function DataRetentionValue({
     value,
     suffix,
+    canRetainForever = true,
     shouldPluralize = true,
 }: DataRetentionValueProps): ReactElement {
     let content = 'Unknown';
 
     if (typeof value === 'number') {
         if (value === 0) {
-            content = 'Never deleted';
+            content = canRetainForever ? 'Never deleted' : 'Deleted in every pruning cycle';
```

Comment:
To take this insightful comment by Van one step farther, duplication of code for **view versus edit** doubles the effort to provide consistent explanation of the meaning of zero.

System Configuration is a specialized page where we need to make changes like this rarely, but usually urgently.

We should explore read-only form controls so the form can also render **view** mode:
https://v4-archive.patternfly.org/v4/components/form-control/design-guidelines#read-only-controls

Reply:
So if I understand correctly, you're suggesting that making the form inputs "read only" in the "view" mode might improve System Config maintainability? Because that makes a lot of sense to me